### ISSUE TITLE
On BulkInsert, Identity Fields Are Not Being Set On The Entity

### ISSUE THREAD

**Issue by DaleyKD** [borisdj/EFCore.BulkExtensions#612]
Is it possible that after doing a Bulk Insert, the Guid ID field will populate its value for the entity?  What about DateTime Created?

``` c#
// My Context
public class MyDbContext : DbContext
{
    protected override void OnModelCreating(ModelBuilder builder)
    {
        // Set the default SQL value used so that SQL will be the one to give a value on INSERT.
        builder.Entity<JobUnit>().Property(ju => ju.ID).HasDefaultValueSql("newsequentialid()");
        builder.Entity<JobUnit>().Property(ju => ju.Created).HasDefaultValueSql("getutcdate()");
    }
}

// My Entity
public class JobUnit
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid ID { get; set; }

    [Required]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public DateTime Created { get; set; }

    [Required]
    public DateTime Modified { get; set; }

    public string Settings { get; set; }
}

// Function that calls the DB service
public async Task Test()
{
    var unit = new JobUnit();
    await jobUnitService.BulkInsertAsync(new[] { unit });
}

// Function in the DB service
public async Task BulkInsertAsync(IList<JobUnit> jobUnits)
{
    // Since we're using EFCore.BulkExtensions, OnBeforeSave doesn't get called on the _ctx
    var dtNow = DateTime.UtcNow;
    foreach (var entity in jobUnits)
    {
        entity.Modified = dtNow;
        entity.Settings = $"EFCore.BulkExtensions: {typeof(DbContextBulkExtensions).Assembly.GetName().Version}";
    }

    await _ctx.BulkInsertAsync(jobUnits, o =>
    {
        o.SetOutputIdentity = true;
    });

    var id = jobUnits.FirstOrDefault()?.ID;
}
```

![image](https://user-images.githubusercontent.com/7206282/128769151-fea0339c-6d1e-4254-ad7a-70d19af51932.png)

**Comment by borisdj** [comment]
Test again with latest v.

**Comment by DaleyKD** [comment]
@borisdj : This is "half" working.  The `public Guid ID` field comes back.  However, `public DateTime Created` doesn't.
![image](https://user-images.githubusercontent.com/7206282/128876559-41ad8dc3-0f72-4722-86fe-e44c4f9b014e.png)

**Comment by borisdj** [comment]
v5.3.7 should fix both halves.

**Comment by DaleyKD** [comment]
My quick test is good to go.  Consider this fixed.  Thank you!

**Comment by borisdj** [comment]
You are welcome.

**Comment by el2iot2** [comment]
@borisdj Is this possibly still an issue in the BulkInsertOrUpdateAsync call as well? We are experiencing the same problem as listed above, but for that call.

**Comment by borisdj** [comment]
I've just run with that method and works fine:
![image](https://user-images.githubusercontent.com/1279604/129163920-4071d645-a489-4b94-a619-9716a822e850.png)
All props are outputed from Db.

**Comment by el2iot2** [comment]
Yes. The behavior I was seeing makes sense after more thought. In an InsertOrUpdateAsync I was expecting IDs to be set for *all* supplied entities. How it actually works is to return IDs for inserted entities and updated entities, but not for any other entities in the supplied list. It took me a bit to realize this is "as designed", behavior and not a bug. Entities with the default id are those that weren't inserted and weren't updated, and this is actually helpful behavior. Thank you for your response.

**Comment by borisdj** [comment]
No problem.

**Comment by AKlaus** [comment]
The issue still exists on the latest version `v8.1.2` against the SQL Server. 
There're some more details in a linked issue #1299 .

**Comment by borisdj** [comment]
New version with latest source are published v 8.1.3 for EF8 and v 9.0.1 for EF9.
Make a test for the issue with these versions.
